AppInspector/MetaDataHelper.cs

```diff
@@ -112,17 +133,43 @@ public void AddMatchRecord(MatchRecord matchRecord)
                 //update list of unique tags as we go
                 foreach (string tag in matchRecord.Tags)
                 {
-                    _ = Metadata.UniqueTags.TryAdd(tag,0);
+                    _ = UniqueTags.TryAdd(tag,0);
                 }
 
-                Metadata.Matches.Add(matchRecord);
+                if (allowAdd)
+                {
+                    Metadata.Matches.Add(matchRecord);
+                }
             }
             else
             {
                 Metadata.IncrementTotalMatchesCount(-1);//reduce e.g. tag counters not included as detailed match
             }
         }
 
+
+        /// <summary>
+        /// Transfer concurrent data from scan to analyze result with sorted, simplier types for callers
+        /// </summary>
+        public void PrepareReport()
```

Comment:
Instead of Metadata being a static class shouldn't this return the serializable `Metadata` object?

Reply:
If I did I'd have to wrap calls for 3 of the data types for incrementing counts which seems needless.

Reply:
I'm not sure I understand why that would involve more call wrapping.

Your analyzer function could in theory be called twice at once and using a static class to hold non-static data like this is just inviting weird behavior.

Reply:
Metadata is not a static class.  If you see it add the code line here.  What I responded interpreted was return a Metadata object from PrepareReport but an instance exists before that and is needed.  Also the AnalyzeCommand should only be called once by a given caller by instance until it completes though the scan is threaded.